packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -452,6 +472,33 @@ export class StateMachine extends StateMachineBase {
       }
     }
 
+    if (props?.kmsDataKeyReusePeriodSeconds && !isValidKmsDataKeyReusePeriodSeconds(props.kmsDataKeyReusePeriodSeconds)) {
+      throw new Error('kmsDataKeyReusePeriodSeconds needs to be a value between 60 and 900');
+    }
+
+    if (props?.kmsKey) {
+      props?.kmsKey.addToResourcePolicy(new iam.PolicyStatement({
+        resources: ['*'],
+        actions: ['kms:Decrypt', 'kms:GenerateDataKey', 'kms:DescribeKey'],
+        principals: [new iam.ServicePrincipal('states.amazonaws.com')],
+        conditions: {
+          StringEquals: {
+            'aws:SourceAccount': this.stack.account,
+            'aws:SourceArn': Stack.of(this).formatArn({
+              service: 'states',
+              resource: 'stateMachine',
+              resourceName: this.physicalName,
+            }),
+            'kms:EncryptionContext:aws:states:stateMachineArn': Stack.of(this).formatArn({
```

Comment:
If the customer is using their own KMS key for StateMachine, they may want their CloudWatch Logs encrypted too or at least have a way to set that up in CDK. The KMS key policy needs to be updated for that

https://docs.aws.amazon.com/step-functions/latest/dg/encryption-at-rest.html#create-state-machine-with-cmk

Reply:
True, currently working on adding such logic.